Libraries/Core/InitializeCore.js

```diff
@@ -44,6 +44,28 @@ if (_shouldPolyfillCollection('Set')) {
   polyfillGlobal('Set', () => require('Set'));
 }
 
+// Wrap Object.{freeze,seal,preventExtensions} so each function adds its
```

Comment:
Can you avoid doing this if the polyfill is not needed (i.e. for versions of JSC where `Map` and `Set` are present)?

Reply:
Also, should this be in a new file/module that lives closer to `Map.js`?

Reply:
Definitely!

Reply:
Improved: 114008536e40c394e990682026d2a8b55922fec6